drivers/metal/rendering_device_driver_metal.mm

```diff
@@ -847,10 +847,13 @@ static const API_AVAILABLE(macos(11.0), ios(14.0)) MTLSamplerBorderColor SAMPLER
 		}];
 	}
 
+	int max_fps = Engine::get_singleton()->get_max_fps();
+	double present_minimum_duration = max_fps ? 1.0 / max_fps : 0.0;
+
```

Comment:
Rather than calling this every time, you will need to override the following for the `MetalRenderingDeviceDriver`:

https://github.com/godotengine/godot/blob/0f5f3bc9546b46b2029fc8896dc859697f1eab97/servers/rendering/rendering_device_driver.h#L461-L463

This can then be cached in the `RenderingContextDriverMetal::Surface`

Reply:
Yeah, that is definitely not your problem, but I expect one we should fix, so your solution can follow the standard approach. How are you setting that value?

Reply:
Thanks – how are you configuring the game to run at 30fps, so I can test it?

Reply:
Thank you!

Reply:
> I tried doing that initially, but there is a separate bug in the engine that prevents me from using that. rd is always nullptr in the following code.
> 
> ```
> void Engine::set_max_fps(int p_fps) {
> _max_fps = p_fps > 0 ? p_fps : 0;
> 
> RenderingDevice *rd = RenderingDevice::get_singleton();
> if (rd) {
> rd->_set_max_fps(_max_fps);
> }
> }
> ```

Where is this code being called? Perhaps you are calling it before the RD is initialilzed?